kfp_notebook/pipeline/_notebook_op.py

```diff
@@ -236,3 +266,40 @@ def _artifact_list_to_str(self, pipeline_array):
                     ValueError("Illegal character ({}) found in filename '{}'.".format(INOUT_SEPARATOR, artifact_name))
             trimmed_artifact_list.append(artifact_name.strip())
         return INOUT_SEPARATOR.join(trimmed_artifact_list)
+
+    @staticmethod
+    def _normalize_label_value(value):
+
+        """Produce a Kubernetes-compliant label from value
+
+        Valid label values must be 63 characters or less and
+        must be empty or begin and end with an alphanumeric
+        character ([a-z0-9A-Z]) with dashes (-), underscores
+        (_), dots (.), and alphanumerics between.
+        """
+
+        if value is None or len(value) == 0:
+            return ''   # nothing to do
+
+        value = value[:63]  # enforce max length 63
+
+        # must begin with [0-9a-zA-Z]
+        valid_chars = string.ascii_letters + string.digits
+        if value[0] not in valid_chars:
+            value = 'a' + value[1:]
+
+        # must end with [0-9a-zA-Z]
+        if value[len(value) - 1] not in valid_chars:
+            value = value[:len(value) - 1] + 'a'
```

Comment:
A more "pythonic" way might be...
```suggestion
        if value[-1] not in valid_chars:
            value = value[:-1] + 'a'
```